packages/adminrouter/extra/src/Makefile.common

```diff
@@ -57,6 +57,7 @@ devkit:
 		echo "Devkit image is present, not rebuilding it."; \
 	else \
 		echo "Building devkit image"; \
+		docker rm -f $(DEVKIT_NAME) ; \
```

Comment:
My understanding from JIRA is that we have not encountered any issues with the container existing at this point. Is this extra caution or is there a case we can get here?

Can we have container `$(DEVKIT_NAME)` without image `mesosphere/$(DEVKIT_NAME)`? If not, is there a race condition between getting to `else` and building something? I'd really like a clarifying comment.

Reply:
> My understanding from JIRA is that we have not encountered any issues with the container existing at this point. Is this extra caution or is there a case we can get here?

This is me being extra cautious. We have not fully understood the root cause, hence we put the removal code even in the places where (in theory) it should not be needed.

> Can we have container $(DEVKIT_NAME) without image mesosphere/$(DEVKIT_NAME)? If not, is there a race condition between getting to else and building something? I'd really like a clarifying comment.

It depends. CI runs both the intermediate container (i.e. `mesosphere/$(DEVKIT_NAME):noresty`) and the final container (i.e. `mesosphere/$(DEVKIT_NAME):full`) as `$(DEVKIT_NAME)`. No, you can't have `$(DEVKIT_NAME)` running without *both* of these images, you can have it if one or the other (or both) have not been properly cleaned up. The `make devkit` target call may happen after the devkit container has been build (or the build was interrupted while the intermediate stage container was still there) as the Jenkins agent can be reused. I am not sure if I answered your question. Please clarify if I did not.

Reply:
> No, you can't have $(DEVKIT_NAME) running without both of these images, you can have it if one or the other (or both) have not been properly cleaned up.

If I understand this correctly, the code says:

```
if Docker image mesosphere/$(DEVKIT_NAME):
   return
else:
   remove container $(DEVKIT_NAME)
```

This can be simplified to:

```
if not Docker image mesosphere/$(DEVKIT_NAME):
    remove container $(DEVKIT_NAME)
```

my question was "can `container $(DEVKIT_NAME)` exist if "Docker image `mesosphere/$(DEVKIT_NAME)`" does not exist.

If not, change adds nothing.

Reply:
Separately, maybe we can do:

```
else
    $(MAKE) update-devkit
```

to reduce duplication / bug surface area.

Reply:
or if not:

```
else
    $(MAKE) clean-devkit-container
```